R/respiratory-condition.R

```diff
@@ -315,7 +318,211 @@ resp_condition_fun3 <-
         (CCC_91A == 2 & CCC_031 == 2)), 3,
     if_else2(
       ((DHHGAGE_cont > 0 & DHHGAGE_cont >= 30) & 
-        (CCC_091 == 2 & CCC_91A == 2 & CCC_031 == 2)), 3, "NA(b)")
+        (CCC_091 == 2 & CCC_91A == 2 & CCC_031 == 2)), 3,
+    if_else2((CCC_091 == "NA(a)" & CCC_91A == "NA(a)" & CCC_031 == "NA(a)"), 
+             "NA(a)", "NA(b)")
+            )
+          )
+        )
+      )
+    )
+  }
+
+#' @title COPD_Emph_der_fun1
+#'
+#' @description This is one of 2 functions used to create a derived variable
+#'  (COPD_Emph_der) that determines if a respondents has either COPD or 
+#'  Emphysema. 2 different functions have been created to account for the fact
+#'  that different respiratory variables are used across CCHS cycles. This
+#'  function is for CCHS cycles (2005-2008) that use COPD and Emphysema as
+#'  a combined variable. 
+#'
+#' @param DHHGAGE_cont continuous age variable.
+#'
+#' @param CCC_91E variable indicating if respondent has Emphysema
+#'
+#' @param CCC_91F variable indicating if respondent has COPD
+#'
+#' @return a categorical variable (COPD_Emph_der) with 3 levels:
+#'
+#'  \enumerate{
+#'  \item respondent is over the age of 35 and has a respiratory condition
+#'  \item respondent is under the age of 35 and has a respiratory condition
+#'  \item respondent does not have a respiratory condition
+#'  }
+#'
+#' @examples
+#' # COPD_Emph_der_fun1() to create values across CCHS cycles
+#' # (2005-2008) COPD_Emph_der_fun1() is specified in
+#' # variable_details.csv along with the CCHS variables and cycles included.
+#'
+#' # To transform COPD_Emph_der, use rec_with_table() for each CCHS cycle
+#' # and specify COPD_Emph_der, along with the various respiratory
+#' # variables. Then by using bind_rows() you can combine COPD_Emph_der
+#' # across cycles.
+#'
+#' library(cchsflow)
+#'
+#' COPD2005 <- suppressWarnings(rec_with_table(
+#'   cchs2005_p,  c(
+#'     "DHHGAGE_cont", "CCC_91E", "CCC_91F",
+#'     "COPD_Emph_der"
+#'   )
+#' ))
+#'
+#' head(COPD2005)
+#'
+#' COPD2007_2008 <- suppressWarnings(rec_with_table(
+#'   cchs2007_2008_p, c(
+#'     "DHHGAGE_cont", "CCC_91E", "CCC_91F", 
+#'     "COPD_Emph_der"
+#'   )
+#' ))
+#'
+#' tail(COPD2007_2008)
+#'
+#' combined_COPD <- suppressWarnings(bind_rows(COPD2005, COPD2007_2008))
+#'
+#' head(combined_COPD)
+#' tail(combined_COPD)
+#' @seealso \code{\link{COPD_Emph_der_fun2}}
+#'
+#' @export
+#' 
+COPD_Emph_der_fun1 <-
+  function(DHHGAGE_cont, CCC_91E, CCC_91F) {
+    # Argument verification
+    if ((!is_equal(CCC_91E, 1) &
+         !is_equal(CCC_91E, 2)) |
+        (!is_equal(CCC_91F, 1) &
+         !is_equal(CCC_91F, 2))) {
```

Comment:
How does this handle CCC_91E or CCC_91F being NA?

Reply:
In addition consider using the %in% function ie: CCC_91E %in% 1:2 | CCC_91F %in% 1:2

Reply:
> How does this handle CCC_91E or CCC_91F being NA?

Line 426 examines if both CCC_91E or CCC_91F are NA(a) to set the function return to NA(a), else NA(b).